File: 0977. Squares of a Sorted Array/Solution.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] res = new int[nums.length];

        int i = 0;
        int j = nums.length - 1;
        int k = nums.length - 1;

        while(k>=0){
            int left = nums[i]*nums[i];
            int right = nums[j]*nums[j];

            if(left>=right){
                res[k] = left;
                k--;
                i++;
            } else {
                res[k] = right;
                k--;
                j--;
            }
        }

        return res;
    }
}

File: 0977. Squares of a Sorted Array/Solution.py
class Solution(object):
    def sortedSquares(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = [0] * len(nums)

        i = 0
        j = len(nums)-1
        k = len(nums)-1


        while(k>=0):
            left = nums[i]*nums[i]
            right = nums[j]*nums[j]

            if left>=right:
                res[k] = left
                i+=1
                k-=1
            else:
                res[k] = right
                j-=1
                k-=1

        return res

